### PR TITLE
unison: reduce closure size when enableX11 = false

### PR DESCRIPTION
**Pull request by hmenke**
## Description of changes

```
Before: /nix/store/ifqr2nhgrci0ixga4k7pk9l6mqdfsk5b-unison-2.53.2        354.3M
After:  /nix/store/b5sn58rrziqrg1bi5m7zisdz785mq45c-unison-2.53.2         40.9M
```

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by hmenke** [comment]
Result of `nixpkgs-review pr 257413` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>unison</li>
    <li>usync</li>
  </ul>
</details>

**Review by Atemu — APPROVED** [review]
Diff LGTM

File: pkgs/applications/networking/sync/unison/default.nix [hunk 1]
```diff
@@ -26,9 +26,10 @@ stdenv.mkDerivation (finalAttrs: {
 
   strictDeps = true;
 
-  nativeBuildInputs = [ glib wrapGAppsHook ocamlPackages.ocaml ]
-    ++ lib.optional enableX11 copyDesktopItems;
-  buildInputs = [ gsettings-desktop-schemas ncurses zlib ]
+  nativeBuildInputs = [ ocamlPackages.ocaml ]
+    ++ lib.optional enableX11 [ copyDesktopItems glib wrapGAppsHook ];
```
Comment (hmenke) [inline pkgs/applications/networking/sync/unison/default.nix:30]: I realized that `glib` can also go under `enableX11`, although there doesn't seem to be any additional reduction in closure size.

Comment (hmenke) [reply pkgs/applications/networking/sync/unison/default.nix:30]: Ah, of course. It's in `nativeBuildInputs`.

Comment (nevivurn) [inline pkgs/applications/networking/sync/unison/default.nix:30]:
```suggestion
    ++ lib.optionals enableX11 [ copyDesktopItems glib wrapGAppsHook ];
```

File: pkgs/applications/networking/sync/unison/default.nix [hunk 2]
```diff
@@ -26,9 +26,10 @@ stdenv.mkDerivation (finalAttrs: {
...
+  buildInputs = [ ncurses zlib ]
+    ++ lib.optional enableX11 [ gsettings-desktop-schemas ]
```
Comment (nevivurn) [inline pkgs/applications/networking/sync/unison/default.nix:32]:
```suggestion
    ++ lib.optional enableX11 gsettings-desktop-schemas
```

**Review by nevivurn — COMMENTED** [review]
Results in slightly different derivations but seems to work either way.

File: pkgs/applications/networking/sync/unison/default.nix (hunk 1 above)
Comment (nevivurn) [reply pkgs/applications/networking/sync/unison/default.nix:30]: Upon closer inspection, unison doesn't require glib and build&works fine without it. Is this correct?

Comment (hmenke) [reply pkgs/applications/networking/sync/unison/default.nix]: Inded, just tried it out and can confirm that `glib` is unnecessary.

**Review by nevivurn — APPROVED** [review]

File: pkgs/applications/networking/sync/unison/default.nix [hunk 3]
```diff
@@ -26,9 +25,10 @@ stdenv.mkDerivation (finalAttrs: {
...
   strictDeps = true;
 
-  nativeBuildInputs = [ glib wrapGAppsHook ocamlPackages.ocaml ]
-    ++ lib.optional enableX11 copyDesktopItems;
-  buildInputs = [ gsettings-desktop-schemas ncurses zlib ]
+  nativeBuildInputs = [ ocamlPackages.ocaml ]
+    ++ lib.optionals enableX11 [ copyDesktopItems wrapGAppsHook ];
+  buildInputs = [ ncurses zlib ]
+    ++ lib.optional enableX11 gsettings-desktop-schemas
     ++ lib.optional stdenv.isDarwin Cocoa;
```
Comment (Atemu) [inline pkgs/applications/networking/sync/unison/default.nix:32]: Nit: Unless there is a deeper meaning behind there being exactly one element in the list, I prefer `optionals [ ... ]`.

**Pull request merged**